tests/integration/utils/routines/data_assertion_destination_routine.py

```diff
@@ -0,0 +1,43 @@
+from multiprocessing import Event
+from pipert2.utils.consts.event_names import KILL_EVENT_NAME
+from pipert2.core.base.routines.destination_routine import DestinationRoutine
+
+
+class DataAssertionDestinationRoutine(DestinationRoutine):
+    """"Routine used to assert that data coming to routine is equal to the expected one.
+    When the data is not equal the routine will set the does_data_equal flag and kill the pipe.
+    When all of the expected data is reached the routine will kill the pipe as well.
+
+    """
+
+    def __init__(self, data_to_expect: list, shared_process_dict: dict,  name="dummy_end_routine"):
+        super(DataAssertionDestinationRoutine, self).__init__(name=name)
+        self.data_to_expect = data_to_expect
+        self.is_data_equal = Event()
```

Comment:
I think is_data_equal doesn't give any information about this parameter. Maybe change this parameter name to is_input_equals_to_expected_input